### DISCUSSION TITLE
What about a no-public-constructor rule?

### DISCUSSION THREAD

**Discussion by sgpinkus** [eslint/eslint#19260] [Ideas] [answered]
Rationale: Javascript unfortunately doesn't support overloaded constructor. But there's often more than one way to construct an instance and having a constructor and then xyz other static factory methods is confusing. Should all just be factories with descriptive names. Plus, having the constructor option even available can lead to trying to shoe horn many different cases into the constructor. I'm not saying it should be on by default.

**Comment by nzakas** [comment, marked as answer]
The most common pattern in JavaScript is to have a public constructor for classes, so I don't think this is a good fit for a core rule.

You can certainly create your own custom rule to do this if you wish.

**Comment by sgpinkus** [comment]
> You can certainly create your own custom rule to do this if you wish.

OK thanks. https://github.com/sgpinkus/eslint-no-public-constructor
